JS/Lesson16/savas/script.js

```diff
@@ -1,75 +1,173 @@
 
-console.log('- - - - - - - - - - - - - - - - - ');
+console.log('- - - - - - - - - - - - - - - - - EXERCISE 9 - - - - - - - - - - - - - - - - - ');
 // 9. Write a do-while loop that counts from 1 to 10 and prints out each number.
+let number = 1;
+do {
+	console.log(number);
+	number++;
+}
+
+while (number <= 10);
 
-console.log('- - - - - - - - - - - - - - - - - ');
+
+console.log('- - - - - - - - - - - - - - - - - EXERCISE 10 - - - - - - - - - - - - - - - - ');
 // 10. Write a for...in loop that prints out the keys and values of an object.
 const person = {name: "John", surname: "Doe", age: 30, city: "New York"};
 // Array olsaydi boyle olurdu: ["John", "Doe", 30 , "New York"];
 // Array element position numbers: 0, 1, 2, 3
 // Object keys: name, surname, age, city 
 
-console.log('- - - - - - - - - - - - - - - - - ');
+for (const objectKey in person) {
+	console.log(`${objectKey}: ${person[objectKey]}`);
+}
+
+
+console.log('- - - - - - - - - - - - - - - - - EXERCISE 11 - - - - - - - - - - - - - - - - ');
 // 11. Write a for...in loop that sums the values of an object.
 const obj = {a: 10, b: 20, c: 30, d: 23};
+let sum = 0;
+for (const key in obj) {
+	sum += obj[key];
+}
+
+console.log(sum);
 
-console.log('- - - - - - - - - - - - - - - - - ');
+console.log('- - - - - - - - - - - - - - - - - EXERCISE 12 - - - - - - - - - - - - - - - - ');
 // 12. Write a loop that finds the average of an array of numbers.
 const numbers12 = [10, 20, 30, 40, 50];
 
-console.log('- - - - - - - - - - - - - - - - - ');
+let sum1 = 0;
+
+for (i = 0; i < numbers12.length; i++) {
+	sum1 += numbers12[i];
+}
+
+let average = sum / numbers12.length;
+
+console.log(`The average of the numbers is ${average}.`);
+
+console.log('- - - - - - - - - - - - - - - - - EXERCISE 13 - - - - - - - - - - - - - - - - ');
 // 13. Write a loop that prints out the sum of all even numbers between 1 and 50.
+let sum2 = 0;
 
+for (i = 0; i <= 50; i += 2) {
+	sum2 += i;
+}
 
+console.log(`The sum of all even numbers between 1 and 50 is ${sum2}.`);
 
-console.log('- - - - - - - - - - - - - - - - - ');
+console.log('- - - - - - - - - - - - - - - - - EXERCISE 14 - - - - - - - - - - - - - - - - ');
 // 14. Write a loop that finds the second largest number in an array of numbers.
-const numbers14 = [10, 5, 20, 8, 15];
-
-console.log('- - - - - - - - - - - - - - - - - ');
+let numbers14 = [10, 5, 20, 8, 15];
+numbers14.sort(function(a, b) {
+	return b - a;
+	}
+);
+console.log(numbers14);
+console.log(`The second largest number is ${numbers14[1]}.`);
+
+console.log('- - - - - - - - - - - - - - - - - EXERCISE 15 - - - - - - - - - - - - - - - - ');
 // 15. Write a loop that counts the number of vowels in a given string.
 const string15 = 'Hello, how are you?';
 const vowels = 'aeiouAEIOU';
+let countsNumVovels = 0;
+for (let i = 0; i < string15.length; i++) {
+	for (let ii = 0; ii < vowels.length; ii++) {
+		if ( string15[i].includes(vowels[ii])) {
```

Comment:
Actually, `vowels.includes(string15[i])` would check if a given character is in vowels so you don't really need the second loop here.

Reply:
I removed the second loop, but I couldn't work it :/

Reply:
The solution is as short as this:
```
for (let i = 0; i < string15.length; i++) {
	if (vowels.includes(string15[i])) countsNumVovels++;
}
console.log(`The number of vowels in "${string15}" is ${countsNumVovels}.`);
```